.github/workflows/main.yml

```diff
@@ -12,7 +12,7 @@ jobs:
     strategy:
       max-parallel: 4
       matrix:
-        java-version: [11, 17, 18]
+        java-version: [17, 18]
```

Comment:
should switch 18 to 21 now maybe?

dropping 11 seems scary imo we should prefer to run an older nessie quarkus server (or make sure even the java11 tests run the server with a newer jdk)

but its your call ultimately

Reply:
Removed because of this error. I saw that Nessie PR builders also work with JDK 17 instead of JDK 11.
Could be something to do with Quarkus. 

```
Execution failed for task ':iceberg-catalog-migrator-api:intTest'.
[325](https://github.com/projectnessie/iceberg-catalog-migrator/actions/runs/8434836997/job/23098937620?pr=148#step:4:327)
> Nessie-Server/Quarkus did not emit listen URL. Process: [/usr/lib/jvm/temurin-11-jdk-amd64/bin/java, -Dquarkus.http.port=0, -Dquarkus.management.port=0, -Dquarkus.log.level=INFO, -Dquarkus.log.console.level=INFO, -jar, /home/runner/.gradle/caches/modules-2/files-2.1/org.projectnessie.nessie/nessie-quarkus/0.79.0/6275b00377182632d42386ad95ebcef148ef1d11/nessie-quarkus-0.79.0-runner.jar]
```

Reply:
tracking via 
https://github.com/projectnessie/iceberg-catalog-migrator/issues/149

Reply:
Definitely against dropping Java 11 here. There are definitely ways to make that work. Just removing Java 11 from CI is hacky IMHO.

Reply:
ok. Using `0.74.0` if the java version is older than 17

Reply:
@snazy: Please let me know if the updated changes are ok.

Reply:
No, not okay. Downgrading is also not an option IMO.

Why don't you run the server w/ Java 17? The plugin supports this, no?

Reply:
> I don't know How the PR builder can automatically install JDK 17.

We are doing this in other builds - maybe look in the docs for the GH action?

Reply:
Which project? I checked Nessie, it was using only JDK 17 and above and couldn't find anything related on query-engine-integration-tests either. 

I configured JDK 17 for all java version related tests. But couldn't specify the nessieQuarkusServer to use JDK 17 always.